openedx/core/djangoapps/schedules/management/commands/tests/test_send_upgrade_reminder.py

```diff
@@ -0,0 +1,257 @@
+import datetime
+import itertools
+from copy import deepcopy
+from unittest import skipUnless
+
+import attr
+import ddt
+import pytz
+from django.conf import settings
+from edx_ace.channel import ChannelType
+from edx_ace.test_utils import StubPolicy, patch_channels, patch_policies
+from edx_ace.utils.date import serialize
+from mock import Mock, patch
+from opaque_keys.edx.keys import CourseKey
+from opaque_keys.edx.locator import CourseLocator
+
+from openedx.core.djangoapps.schedules import resolvers, tasks
+from openedx.core.djangoapps.schedules.management.commands import send_upgrade_reminder as reminder
+from openedx.core.djangoapps.schedules.tests.factories import ScheduleConfigFactory, ScheduleFactory
+from openedx.core.djangoapps.site_configuration.tests.factories import SiteConfigurationFactory, SiteFactory
+from openedx.core.djangolib.testing.utils import CacheIsolationTestCase, skip_unless_lms
+from student.tests.factories import UserFactory
+
+
+@ddt.ddt
+@skip_unless_lms
+@skipUnless('openedx.core.djangoapps.schedules.apps.SchedulesConfig' in settings.INSTALLED_APPS,
+            "Can't test schedules if the app isn't installed")
+class TestUpgradeReminder(CacheIsolationTestCase):
```

Comment:
Maybe this will wait until the later refactor, but it seems like we shouldn't need separate copies of the tests to test the two different types of reminders, since they both use the same internal logic.

Reply:
Yeah, I'm inclined to push this off until later so we can get this merged. I added a task to the wiki.